internal/fields/validate.go

```diff
@@ -354,60 +354,74 @@ func compareKeys(key string, def FieldDefinition, searchedKey string) bool {
 }
 
 func (v *Validator) parseElementValue(key string, definition FieldDefinition, val interface{}) error {
-	val, ok := ensureSingleElementValue(val)
-	if !ok {
-		return nil // it's an array, but it's not possible to extract the single value.
-	}
+	return forEachElementValue(val, func(val interface{}) error {
+		var valid bool
+		switch definition.Type {
+		case "constant_keyword":
+			var valStr string
+			valStr, valid = val.(string)
+			if !valid {
+				break
+			}
 
-	var valid bool
-	switch definition.Type {
-	case "constant_keyword":
-		var valStr string
-		valStr, valid = val.(string)
-		if !valid {
-			break
-		}
+			if err := ensureConstantKeywordValueMatches(key, valStr, definition.Value); err != nil {
```

Comment:
It looks like the switch evolved since I remember :) Do you think we could transform the field type-driven logic into properties-driven? I mean, for every field we could check keyword, pattern, allowed values, and simply skip testing if not applicable. Otherwise, this switch will grow up indefinitely...

Reply:
Umm, we still need custom logic per type:
* Only for `constant_keyword` we check the value is the defined one.
* Only for `ip` we check that is one of the allowed ips.
* Only for `numbers` we check that they are actually numbers.

To skip testing when not applicable we would still need some conditional logic per type.

I think the switch fits well for this. One thing I can do is to group back `keyword`, `text` and `date`. Though I am not sure why `date` was checked as if were `text`.

Reply:
> I think the switch fits well for this. One thing I can do is to group back keyword, text and date. Though I am not sure why date was checked as if were text.

I don't remember now to be honest. Maybe @adriansr remembers this?

Reply:
Handling of `date` changed more in https://github.com/elastic/elastic-package/pull/781, so I guess that this is going to continue being separated.